benchmark/FedHPOB/fedhpob/optimizers/grid_search.py

```diff
@@ -0,0 +1,64 @@
+# Implement GS with `ConfigSpace.util.generate_grid`
+
+import time
+import random
+import logging
+import ConfigSpace as CS
+from ConfigSpace.util import generate_grid
+
+from fedhpob.config import fhb_cfg
+from fedhpob.utils.monitor import Monitor
+
+logging.basicConfig(level=logging.WARNING)
+
+
+def run_grid_search(cfg):
+    monitor = Monitor(cfg)
+    cfg = cfg.clone()
+    benchmark = cfg.benchmark.cls[0][cfg.benchmark.type](
+        cfg.benchmark.model,
+        cfg.benchmark.data,
+        cfg.benchmark.algo,
+        device=cfg.benchmark.device)
+    # configspace = cfg.benchmark.configuration_space[0]
+    # This is only for FEMNIST grid search exp
+    new_configspace = CS.ConfigurationSpace()
+    new_configspace.add_hyperparameter(
+        CS.CategoricalHyperparameter('batch', choices=[16, 64]))
+    new_configspace.add_hyperparameter(
+        CS.CategoricalHyperparameter('dropout', choices=[0.0, 0.5]))
+    new_configspace.add_hyperparameter(
+        CS.CategoricalHyperparameter(
+            'lr', choices=[0.01, 0.02783, 0.07743, 0.21544, 0.59948, 1.0]))
+    new_configspace.add_hyperparameter(
+        CS.CategoricalHyperparameter('step', choices=[2, 4]))
+    new_configspace.add_hyperparameter(
+        CS.CategoricalHyperparameter('wd', choices=[0.0, 0.1]))
+    grid = generate_grid(new_configspace)
```

Comment:
I cannot find another argument to specify the numbers of samples.

Reply:
Since `generate_grid` cannot handle category hyperparam, the `new_configspace ` in Line 37 has been reduced to grid manually.